content/doc/tutorials/build-a-cpp-app-with-jenkins.adoc

```diff
@@ -0,0 +1,377 @@
+
+Selecting the *Build* stage will provide more information about the stage, including the output of the `cmake` command if you select the green `cmake` section.
+
+[.boxshadow]
+image:tutorials/cpp/pipeline-console-build-stage.png[alt="Build stage",width=100%]
+
+You can now select *casino-number-guessing-game* (if that's the name you chose for your pipeline) on the top left to return to your pipeline main page.
+
+=== Add a test stage to your Pipeline
+
+. Go back to your text editor/IDE and ensure your `Jenkinsfile` is open.
+. Copy and paste the following Declarative Pipeline syntax immediately under the `Build` stage:
++
+[source,groovy]
+----
+        stage('Test') {
+            steps {
+                sh './build/casino_game'
+                sh './build/test_game'
+            }
+        }
+----
+so that you end up with:
++
+[source,groovy]
+----
+pipeline {
+    agent any
+    stages {
+        stage('Build') {
+            steps {
+                sh 'rm -rf build'
+                sh 'cmake -B build -S .'
+                sh 'cmake --build build'
+            }
+        }
+        stage('Test') { // <1>
+            steps {
+                sh './build/casino_game' // <2>
+                sh './build/test_game'
+            }
+        }
+    }
+}
+----
+<1> Defines a link:/doc/book/pipeline/syntax/#stage[`stage`] (directive) called `Test` that appears on the Jenkins UI.
+<2> This link:/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script[`sh`] step runs the game and unit tests to ensure the application works as expected.
+
+. Save your edited `Jenkinsfile` and commit it to your local `casino-number-guessing-game` Git repository. Within the `casino-number-guessing-game` directory, run the commands: +
+  `git add Jenkinsfile` +
+  then +
+  `git commit -m "Add 'Test' stage"` +
+  and finally +
+  `git push` to push your changes to your forked repository on GitHub, so it can be picked up by Jenkins.
+
+. In Jenkins, go back to your *Dashboard* if necessary, then select *casino-number-guessing-game* and launch another build thanks to *Build Now*.
+. After a while, a new column titled *Test* will appear in the *Stage View*.
+
+[.boxshadow]
+image:tutorials/cpp/test-stage-overview.png[alt="Test stage",width=100%]
+
+You can select *#2*, or on the number representing your last build on the left, under *Build History* to see the details of the build.
+
+[.boxshadow]
+image:tutorials/cpp/test-stage-status.png[alt="Test stage",width=100%]
+
+You can now select *Pipeline Overview* to see the stages of the Pipeline.
+
+[.boxshadow]
+image:tutorials/cpp/pipeline-overview-test-stage.png[alt="Pipeline overview with Test stage",width=100%]
+
+The newly added "Test" stage is visible here. Selecting the "Test" stage checkmark displays the stage output.
+
+[.boxshadow]
+image:tutorials/cpp/pipeline-output-test-stage.png[alt="Test stage output",width=100%]
+
+==== Viewing the Results
```

Comment:
```suggestion
==== Viewing the results
```
Suggestion to use sentence casing (only beginning of header and any propper nouns).